Fix flaky test testToJSONString

219sansim opened the pull request:
### Problem
The `testToJSONString` is detected flaky because the `JsonHelper.toJSONString()` gives a non-deterministic output as JSONObject values are unordered. 

### Solution
After modifying `testToJSONString` to expect 2 different string values, the test passes.

**Reason:**
The JSON object is unordered and generally uses Maps as their internal data structure. And, according to [HashMap documentation](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html),
_"This class makes no guarantees as to the order of the map; in particular, it does not guarantee that the order will remain constant over time."_
This causes the `JsonHelper.toJSONString()` to give a non-deterministic output.

### Reproduction of the error
The test fails on 5/10 runs of the [NonDex](https://github.com/TestingResearchIllinois/NonDex) tool

**Command to reproduce the failure:**
```mvn -pl pac4j-oauth edu.illinois:nondex-maven-plugin:2.1.7-SNAPSHOT:nondex -Dtest=org.pac4j.oauth.profile.JsonHelperTests#testToJSONString -DnondexRuns=10```

 **Error Message:**
```[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.410 s <<< FAILURE! -- in org.pac4j.oauth.profile.JsonHelperTests
[ERROR] org.pac4j.oauth.profile.JsonHelperTests.testToJSONString -- Time elapsed: 0.396 s <<< FAILURE!
org.junit.ComparisonFailure: expected:<"{\"[id\":\"id\",\"name\":\"name]\"}"> but was:<"{\"[name\":\"name\",\"id\":\"id]\"}">
```

Commit:
Fix flaky test testToJSONString

hofi1 commented:
Personal Preference: 
Split your PR into 3 major section: 

## Problem
## Solution
## Reproduction of the error

Split the overview into these section, so it states clear what is the current state and what needs to be changed to fix the issue.

Furthermore, your solution looks good. Maybe you can find a better assert statement than that one containing the logical or.

Beside that, LGTM!

harshith2000 commented:
Hi! 
While the fix looks good to me, your PR description can be better. Please follow the structure mentioned by hofi1.

219sansim commented:
> Personal Preference: Split your PR into 3 major section:
> 
> ## Problem
> ## Solution
> ## Reproduction of the error
> Split the overview into these section, so it states clear what is the current state and what needs to be changed to fix the issue.
> 
> Furthermore, your solution looks good. Maybe you can find a better assert statement than that one containing the logical or.
> 
> Beside that, LGTM!

Hi I have changed the PR description and I checked for ways to remove the logical or in the assert statement. But it seems it would require me to add dependencies on the project. Source: [stackoverflow question](https://stackoverflow.com/questions/39179125/what-is-assertthat-method)

hofi1 commented:
well done, LGTM!

zzjas commented:
You can proceed to open a real PR. Once you open a real PR, please mark this tentative PR as `Opened` in your `tentative_pr.csv` file and also raise a PR to IDoFT marking this as `Opened`. Thanks!
